warn/warn_control_flow.go

```diff
@@ -376,14 +376,33 @@ func findUninitializedVariables(stmts []build.Expr, previouslyInitialized map[st
 	// findUninitializedIdents traverses an expression (simple statement or a part of it), and calls
 	// `callback` on every *build.Ident that's not mentioned in the map of initialized variables
 	findUninitializedIdents := func(expr build.Expr, callback func(ident *build.Ident)) {
+		// Collect lValues, they shouldn't be taken into account
+		// For example, if the expression is `a = foo(b = c)`, only `c` can be an unused variable here.
+		lValues := make(map[*build.Ident]bool)
 		build.Walk(expr, func(expr build.Expr, stack []build.Expr) {
-			if ident, ok := expr.(*build.Ident); ok && !initialized[ident.Name] {
+			if bin, ok := expr.(*build.BinaryExpr); ok && bin.Op == "=" {
```

Comment:
It would be useful to have an AssignStmt in the future, to avoid this kind of confusion.

Reply:
I agree, but this case also covers assignment expressions (as for keyword arguments). Buildifier should warn about uninitialized `x` in `f(x = y)`.

Perhaps I'll create an AssignmentExpr node (we don't have separate statements nodes anyway).

Reply:
In `f(x = y)`, we shouldn't warn about x, only y.